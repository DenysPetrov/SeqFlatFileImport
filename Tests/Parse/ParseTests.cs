using System.Runtime.CompilerServices;
using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;
using SeqFlatFileImport.Tests.Helpers;

namespace SeqFlatFileImport.Tests.Parse
{
    public class ParseTests
    {
        [Test]
        [UseReporter(typeof(BeyondCompareReporter))]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void OctopusServer()
        {
            Execute("OctopusServer.txt");
        }

        [Test]
        [UseReporter(typeof(BeyondCompareReporter))]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void OctopusWebLog()
        {
            Execute("Web-2017-01-18.log");
        }


        [Test]
        [UseReporter(typeof(BeyondCompareReporter))]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void Iis()
        {
            Execute("u_ex170117.log");
        }

        [Test]
        [UseReporter(typeof(BeyondCompareReporter))]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void OctopusTask()
        {
            Execute("ServerTasks-16572.log.txt");
        }

        [Test]
        [UseReporter(typeof (BeyondCompareReporter))]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void OctopusRawTask()
        {
            Execute("servertasks-21_gjzwyyv2kt.txt");
        }

        public void Execute(string inputFileName)
        {
            var seqServer = new StubSeqEndpoint();
            var result = new Importer(seqServer)
                .Import(TestHelper.GetFilePath(inputFileName));
            result.ShouldBeSuccessful();
            Approvals.VerifyJson(seqServer.LogsAsJson);
        }


    }
}