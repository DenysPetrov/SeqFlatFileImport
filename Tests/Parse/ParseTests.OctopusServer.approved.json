[
    {
        "Timestamp": "2016-05-30T00:00:00.1811+10:00",
        "Level": "Information",
        "MessageTemplate": "Reader took 353ms (319ms until the first record): SELECT * FROM dbo.[Dashboard] ORDER BY Id\r\n2016-05-30 08:49:44.9164   2283 ERROR  Unhandled exception from web server: The semaphore timeout period has expired",
        "Properties": {
            "LineNumber": 1,
            "Thread": "2749"
        },
        "Exception": "System.Net.HttpListenerException (0x80004005): The semaphore timeout period has expired   at System.Net.HttpResponseStream.Write(Byte[] buffer, Int32 offset, Int32 size)   at System.IO.Compression.DeflateStream.PurgeBuffers(Boolean disposing)   at System.IO.Compression.DeflateStream.Dispose(Boolean disposing)   at System.IO.Stream.Close()   at System.IO.Compression.GZipStream.Dispose(Boolean disposing)   at System.IO.Stream.Close()   at Octopus.Server.Web.Infrastructure.NancyCompression.<>c__DisplayClass3_0.<CompressResponse>b__0(Stream responseStream) in Y:\\work\\refs\\tags\\3.3.15\\source\\Octopus.Server\\Web\\Infrastructure\\NancyCompression.cs:line 100   at Octopus.Server.Web.OctopusNancyHost.OutputWithDefaultTransferEncoding(Response nancyResponse, HttpListenerResponse response) in Y:\\work\\refs\\tags\\3.3.15\\source\\Octopus.Server\\Web\\OctopusNancyHost.cs:line 350   at Octopus.Server.Web.OctopusNancyHost.ConvertNancyResponseToResponse(NancyContext nancyRequest, Response nancyResponse, HttpListenerResponse response) in Y:\\work\\refs\\tags\\3.3.15\\source\\Octopus.Server\\Web\\OctopusNancyHost.cs:line 338   at Nancy.NancyEngineExtensions.HandleRequest(INancyEngine nancyEngine, Request request, Func`2 preRequest, Action`1 onComplete, Action`1 onError, CancellationToken cancellationToken)   at Octopus.Server.Web.OctopusNancyHost.Process(HttpListenerContext ctx) in Y:\\work\\refs\\tags\\3.3.15\\source\\Octopus.Server\\Web\\OctopusNancyHost.cs:line 0"
    },
    {
        "Timestamp": "2016-05-30T08:49:46.0918+10:00",
        "Level": "Information",
        "MessageTemplate": "poll://b2yf9fvn5hggm5hvjz72/   1717  Request IScriptService::GetStatus[610] / d5d0972f-1db3-4cde-ac71-5e42c9114040 was queued",
        "Properties": {
            "LineNumber": 15,
            "Thread": "1717"
        },
        "Exception": null
    },
    {
        "Timestamp": "2016-05-30T08:49:46.0918+10:00",
        "Level": "Information",
        "MessageTemplate": "poll://b2yf9fvn5hggm5hvjz72/   1717  Request IScriptService::GetStatus[610] / d5d0972f-1db3-4cde-ac71-5e42c9114040 was collected by the polling endpoint",
        "Properties": {
            "LineNumber": 16,
            "Thread": "1717"
        },
        "Exception": null
    },
    {
        "Timestamp": "2016-05-30T08:49:46.1074+10:00",
        "Level": "Information",
        "MessageTemplate": "Reader took 332ms (332ms until the first record): SELECT * FROM dbo.[Dashboard] WHERE (ProjectId = @projectId) ORDER BY Id",
        "Properties": {
            "LineNumber": 17,
            "Thread": "1716"
        },
        "Exception": null
    },
    {
        "Timestamp": "2016-05-30T08:49:46.2636+10:00",
        "Level": "Information",
        "MessageTemplate": "Reader took 435ms (279ms until the first record): SELECT * FROM dbo.[Dashboard] ORDER BY Id",
        "Properties": {
            "LineNumber": 18,
            "Thread": "2426"
        },
        "Exception": null
    },
    {
        "Timestamp": "2016-05-30T08:49:46.4512+10:00",
        "Level": "Information",
        "MessageTemplate": "listen://[::]:10943/           1056  Accepted TCP client: [::ffff:82.208.129.105]:63248",
        "Properties": {
            "LineNumber": 19,
            "Thread": "1056"
        },
        "Exception": null
    },
    {
        "Timestamp": "2016-05-30T08:49:46.4512+10:00",
        "Level": "Information",
        "MessageTemplate": "listen://[::]:10943/           1062  Performing TLS server handshake",
        "Properties": {
            "LineNumber": 20,
            "Thread": "1062"
        },
        "Exception": null
    },
    {
        "Timestamp": "2016-05-30T08:49:46.498+10:00",
        "Level": "Debug",
        "MessageTemplate": "listen://[::]:10943/           1062  Secure connection established, client is not yet authenticated, client connected with Tls12",
        "Properties": {
            "LineNumber": 21,
            "Thread": "1062"
        },
        "Exception": null
    },
    {
        "Timestamp": "2016-05-30T08:49:46.5918+10:00",
        "Level": "Information",
        "MessageTemplate": "listen://[::]:10943/           1062  A client at [::ffff:82.208.129.105]:63248 connected, and attempted a message exchange, but it presented a client certificate with the thumbprint '644500E8DEB23FA2FE6E009897A0DC4CA51B07BF' which is not in the list of thumbprints that we trust",
        "Properties": {
            "LineNumber": 22,
            "Thread": "1062"
        },
        "Exception": null
    },
    {
        "Timestamp": "2016-05-30T08:49:46.6856+10:00",
        "Level": "Information",
        "MessageTemplate": "poll://b2yf9fvn5hggm5hvjz72/   2413  Request IScriptService::GetStatus[177] / 25f849d3-fc86-476c-ba02-de8eeb3566dc was queued",
        "Properties": {
            "LineNumber": 23,
            "Thread": "2413"
        },
        "Exception": null
    },
    {
        "Timestamp": "2016-05-30T08:49:46.6856+10:00",
        "Level": "Information",
        "MessageTemplate": "poll://b2yf9fvn5hggm5hvjz72/   2413  Request IScriptService::GetStatus[177] / 25f849d3-fc86-476c-ba02-de8eeb3566dc was collected by the polling endpoint",
        "Properties": {
            "LineNumber": 24,
            "Thread": "2413"
        },
        "Exception": null
    },
    {
        "Timestamp": "2016-05-30T08:49:46.7949+10:00",
        "Level": "Information",
        "MessageTemplate": "listen://[::]:10943/           2424  Accepted TCP client: [::ffff:40.68.217.115]:1215",
        "Properties": {
            "LineNumber": 25,
            "Thread": "2424"
        },
        "Exception": null
    },
    {
        "Timestamp": "2016-05-30T12:09:20.3683+10:00",
        "Level": "Information",
        "MessageTemplate": "Reader took 406ms (362ms until the first record): SELECT * FROM dbo.[Dashboard] ORDER BY Id",
        "Properties": {
            "LineNumber": 26,
            "Thread": "412"
        },
        "Exception": null
    },
    {
        "Timestamp": "2016-05-30T12:09:20.4621+10:00",
        "Level": "Information",
        "MessageTemplate": "Request took 1320ms: POST http://octopus.veinteractive.com/api/users/login",
        "Properties": {
            "LineNumber": 27,
            "Thread": "38"
        },
        "Exception": null
    },
    {
        "Timestamp": "2016-05-30T12:09:22.8216+10:00",
        "Level": "Information",
        "MessageTemplate": "listen://[::]:10943/            530  A client at [::ffff:40.68.217.115]:1198 connected, and attempted a message exchange, but it presented a client certificate with the thumbprint '5A53F4E04DE1216DFCAD66735FD04120DB70C74D' which is not in the list of thumbprints that we trust",
        "Properties": {
            "LineNumber": 28,
            "Thread": "530"
        },
        "Exception": null
    }
]